source/defclass-star.lisp

```diff
@@ -302,22 +383,30 @@ The predicate function returns true when the argument is a type of the `name' cl
                  ,@(when (or *symbols-to-export*
                              *export-class-name-p*)
                      `((eval-when (:compile-toplevel :load-toplevel :execute)
-                         (export '(,@(append (when *export-class-name-p*
-                                         (list name))
-                                       *symbols-to-export*))
+                         ;; Don't try to export symbols that don't belong to *package*.
+                         ;; This can happen when inheriting from a class and
+                         ;; overriding some slot.
+                         (export '(,@(remove-if (lambda (sym)
+                                                  (not (eq (symbol-package sym) *package*)))
```

Comment:
But wait, `*package*` may be destructively modified (as it happens in Nyxt), does this clause process it properly? I'd personally rather look as the package of class name instead, but that's not exactly bullet-proof either...

Reply:
Can you give an example?

This export work for slots written like this:

```lisp
(define-class foo ()
  ((other-package::MY-SLOT ...))
```

Reply:
Ah no wait, we no longer `setf *package*` in Nyxt. Then there's no problem, yes!